ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_isis.py

```diff
@@ -73,15 +73,21 @@ def router_isis(self) -> dict | None:
 
     @cached_property
     def _isis_net(self) -> str | None:
-        isis_system_id_prefix = get(self.shared_utils.switch_data_combined, "isis_system_id_prefix")
-        if isis_system_id_prefix is not None:
-            isis_area_id = get(self._hostvars, "isis_area_id", default="49.0001")
-            if self.shared_utils.id is None:
-                raise AristaAvdMissingVariableError(f"'id' is not set on '{self.shared_utils.hostname}' and is required to set ISIS NET address using prefix")
+        if "node_id" == get(self._hostvars, "isis_system_id_format", default="node_id"):
+            isis_system_id_prefix = get(self.shared_utils.switch_data_combined, "isis_system_id_prefix")
+            if isis_system_id_prefix is None:
+                return None
```

Comment:
This TODO is not for your PR, but just something we need to improve some time in the future (might be a breaking change).
```suggestion
                # TODO: Raise for this situation if underlay is ISIS.
                return None
```